integration/test/SetupHelpers.hs

```diff
@@ -163,6 +166,15 @@ createMLSOne2OnePartner domain other convDomain = loop
         then pure u
         else loop
 
+-- Copied from `src/CargoHold/API/V3.hs` and inlined to avoid pulling in `types-common`
+randomToken :: HasCallStack => App String
+randomToken = map mkUrlSafe . unpack . B64Url.encode <$> liftIO (getRandomBytes 16)
+  where
+    mkUrlSafe :: Char -> Char
+    mkUrlSafe '/' = '_'
+    mkUrlSafe '+' = '-'
+    mkUrlSafe c = c
+
```

Comment:
You can use the URL-safe version of Base64 directly: https://hoogle.zinfra.io/file/nix/store/cq3gj4j8yicrmwgl5gr2jak51qg6p6il-base64-bytestring-1.2.1.0-doc/share/doc/base64-bytestring-1.2.1.0/html/Data-ByteString-Base64-URL.html.

Reply:
Ah nice. Don't know how I missed that one.